### ISSUE TITLE
Timestamp value conversion in Cloud Datastore connector is wrong

### ISSUE THREAD

**Issue by damccorm** [apache/beam#21660]
Code in org.apache.beam.sdk.io.gcp.datastore.EntityToRow class:
```

com.google.protobuf.Timestamp time = val.getTimestampValue();
long millis = time.getSeconds() * 1000
+ time.getNanos() / 1000;
return Instant.ofEpochMilli(millis).toDateTime();
```

1 millisecond = 1,000,000 nanoseconds

Imported from Jira [BEAM-14541](https://issues.apache.org/jira/browse/BEAM-14541). Original Jira may contain additional context.
Reported by: yixiaoshen.

**Comment by damccorm** [comment]
Unable to assign user @yixiaoshen. If able, self-assign, otherwise tag @damccorm so that he can assign you. Because of GitHub's spam prevention system, your activity is required to enable assignment in this repo.

**Comment by damccorm** [comment]
Done - FWIW once https://github.com/apache/beam/pull/21719 is merged, you'll be able to self-assign by commenting `.take-issue`

**Comment by yixiaoshen** [comment]
This is fixed by #17789, but I don't seem to have permission to .close-issue

**Comment by damccorm** [comment]
> This is fixed by https://github.com/apache/beam/pull/17789, but I don't seem to have permission to .close-issue

You just did 😃 anyone can do it by commenting `.close-issue` like you just did
